src/probe_modules/module_dns.c

```diff
@@ -1041,7 +1042,23 @@ void dns_process_packet(const u_char *packet, uint32_t len, fieldset_t *fs,
 			}
 			// Did we parse OK?
 			fs_add_uint64(fs, "dns_parse_err", err);
+
+			int idx = fs_get_index_by_name(fs, "dns_answers");
+			if (idx >= 0) {
+				list = fs_get_fieldset_by_index(fs, idx);
+				idx = fs_get_index_by_name(list, "rdata_fs");
+				if (idx >= 0) {
+					list = fs_get_fieldset_by_index(list, idx);
+					idx = fs_get_index_by_name(list, "rdata");
```

Comment:
Do these idx values change record-by-record? If not, seems like you can bypass quite a lot of log(n) lookups via fs_get_index_by_name()